LogfileIterator/LogfileIteratorExample.cpp

```diff
@@ -0,0 +1,179 @@
+ *
+ * @file LogfileIteratorExample.cpp
+ * @brief LogfileIterator Source
+ *
+ */
+
+
+#include "LogfileTestNode.hpp"             // For all CPP Logfile API examples.
+
+using namespace std;
+
+using namespace polysync::datamodel;
+
+
+void LogFileTestNode::prepareLogfileToIterate()
+{
+    sleepMicro ( 1000000 ); //always sleep before modeoff so queue doesn't flush.
+
+    _logFile->setModeOff(); // automatically disables state.
+
+    _logFile->setFilePath( "/tmp/polysync_logfile.plog"  );
+
+    cout<<"Logfile Iterator started." << endl;
+
+    _logFileWasIterated = true;
+}
+
+
+void LogFileTestNode::logfileIteratorCallback(
+        const ps_logfile_attributes * const fileAttributes,
+        const ps_msg_type msg_type,
+        const ps_rnr_log_record * const logRecord,
+        void * const userData )
+{
+    cout<<"\nLogfile iterator callback\n";
+
+    if( logRecord )
+    {
+        cout<<"Log Record: Index: " << logRecord->index <<" - Size: "
+           << logRecord->size <<" - Previous Size: "
+           << logRecord->prev_size <<" - RnR Timestamp ( \nheader.timestamp):"
+           << logRecord->timestamp;
+    }
+
+    cout<<endl;
+
+    // get ps_msg_ref, and do something with it.
+}
+
+
+
+void LogFileTestNode::iterateOverLogfiles()
+{
+    _logFile->forEachIterator(
+                _logFile->getFilePath().c_str(),
+                &logfileIteratorCallback,
+                nullptr );
+}
+
+
+void LogFileTestNode::printResults()
+{
+    if( _logFileWasIterated )
+    {
+        cout<<"\nIterator complete.\n";
+    }
+
+    cout<<"\nReleasing logfile resources. If all messages did not print "
+          "to Terminal \nin either read, write, or iterator, that is due to "
+          "i/o (printf / cout) \nbeing slower than CPU.\n\n";
+}
+
+
+void LogFileTestNode::initStateEvent()
+{
+// 1. Init LogFile API resources:
+    _logFile = new Logfile{ *this };
+
+// 2. Set up parameters.
+    prepareLogfileToIterate();
+
+// 3. Iterate over Logfiles in initStateEvent(), not okStateEvent(), as
+//    Logfile iterator operates outside of the normal Replay time domain.
+    iterateOverLogfiles();
+
+// 4. Disconnect after Iterator is done iterating over a Logfile.
+    disconnectPolySync();
+}
+
+
+void LogFileTestNode::okStateEvent()
+{
+/**** Logfile Iterator. No code should be in in okStateEvent(). ****/
+}
+
+
+void LogFileTestNode::releaseStateEvent()
+{
+    printResults();
+
+// Turn off mode. Turning off the mode automatically disables state.
+// Sleep before turning mode off after last write to avoid flushing of queue.
+
+    sleep( 5 );
+
+    _logFile->setModeOff();
+
+    delete _logFile;
+}
+
+
+int main()
+{
+    cout<<"\n\n************************************************\n"
+    "*** PolySync LogFile API: C++ Iterator Example ***\n\n";
+
+    try
+    {
+        LogFileTestNode aNode;
+
+        sleep( 2 );
+
+// Regarding setting node name using Logfile Iterator and how that relates to
```

Comment:
I'd prefer these to be block comments.

/* 
 * Regarding setting node name using Logfile Iterator and how that relates to
 * setFilePath(), setSessionId(), and setNodeName():
 * 
 * Example: in directory 1234, let a logfile be name.5678.plog,
 * where 1234 = session ID, 5678 = GUID, and name = node name.
 * 
 * To iterate over records in a logfile, use its explicit filepath.
 * 
 * Any file specified in the path can be read from; node name is irrelevant.
 * How: use any setNodeName(), and use setFilePath() in prepareLogfileToIterate().
 */